src/haystack/val/number.rs

```diff
@@ -126,21 +126,31 @@ impl Hash for Number {
 
 impl PartialEq for Number {
     fn eq(&self, other: &Self) -> bool {
-        self.value.eq(&other.value) && self.unit.eq(&other.unit)
+        self.value == other.value && self.unit == other.unit
     }
 }
 
 impl Eq for Number {}
 
 impl PartialOrd for Number {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(self.value.to_bits().cmp(&other.value.to_bits()))
+        if self.unit == other.unit {
+            Some(self.cmp(other))
```

Comment:
`cmp` is meant for types that form a [total order](https://en.wikipedia.org/wiki/Total_order) which is not the case for floating numbers. In fact `f64` does not implement `Ord` due to `NaN`.  `f64` can still be compared with `PartialOrd`, so this could be changed to:
```rust
if self.unit == other.unit {
    self.value.partial_cmp(&other.value)
} 
``` 
What do you think?

Reply:
This is a good question, `Ord` is required for the whole `Value` type, so we can use it as key in a `Map`, or if used in a `Vector`, the vector can be sorted.

The proposed change should cover the NAN case and behave similarly to the Haystack core one (I need to fix the order for non-equal returns).https://github.com/j2inn/haystack-core/blob/master/src/core/HNum.ts#L331

Reply:
We could technically deviate from that behavior and `self.value.partial_cmp(&other.value)` with NaNs would always be  `None` such as `a != b` where `a` and `b` are NaNs.
With the proposed change this is true `a == b`

Reply:
OK, thinking more about this, I will go with the expected Rust behaviour here and rely on `partial_cmp`